lib/node_modules/@stdlib/stats/base/dcumax/lib/dcumax.js

```diff
@@ -41,58 +41,17 @@ var isPositiveZero = require( '@stdlib/math/base/assert/is-positive-zero' );
 *
 * var x = new Float64Array( [ 1.0, -2.0, 2.0 ] );
 * var y = new Float64Array( x.length );
-* var N = x.length;
 *
-* var v = dcumax( N, x, 1, y, 1 );
+* var v = dcumax( x.length, x, 1, y, 1 );
 * // returns <Float64Array>[ 1.0, 1.0, 2.0 ]
 */
 function dcumax( N, x, strideX, y, strideY ) {
-	var max;
-	var ix;
-	var iy;
-	var v;
-	var i;
+	var ox;
+	var oy;
 
-	if ( N <= 0 ) {
-		return y;
-	}
-	if ( strideX < 0 ) {
-		ix = (1-N) * strideX;
-	} else {
-		ix = 0;
-	}
-	if ( strideY < 0 ) {
-		iy = (1-N) * strideY;
-	} else {
-		iy = 0;
-	}
-	max = x[ ix ];
-	y[ iy ] = max;
-
-	iy += strideY;
-	i = 1;
-	if ( isnan( max ) === false ) {
-		for ( i; i < N; i++ ) {
-			ix += strideX;
-			v = x[ ix ];
-			if ( isnan( v ) ) {
-				max = v;
-				break;
-			}
-			if ( v > max || ( v === max && isPositiveZero( v ) ) ) {
-				max = v;
-			}
-			y[ iy ] = max;
-			iy += strideY;
-		}
-	}
-	if ( isnan( max ) ) {
-		for ( i; i < N; i++ ) {
-			y[ iy ] = max;
-			iy += strideY;
-		}
-	}
-	return y;
+	ox = stride2offset( N, strideX );
```

Comment:
Feel free to declare and assign here, rather than declare and then assign later.